recipes-omnect/initrdscripts/omnect-os-initramfs/omnect-device-service-setup

```diff
@@ -19,47 +19,70 @@ get_id() {
+    run_cmd chmod 600 ${omnect_run_dir}/fsck.json || return 1
+
+    return 0
+}
+
 # setup conditional temp files for omnect-device-service, e.g. for the update workflow handling;
 # (static temp files for omnect-device-service are created via systemd-tmpfiles.d)
 run_omnect_device_service_setup () {
-    local omnect_validate_update_group=""
-    local omnect_device_service_user=""
+    local omnect_device_service_group="$(get_id omnect_device_service ${ROOTFS_DIR}/etc/group)"
+    local omnect_device_service_user="$(get_id omnect_device_service ${ROOTFS_DIR}/etc/passwd)"
     local omnect_run_dir="${ROOTFS_DIR}/run/omnect-device-service"  # /run is available in the INITRAMFS
     local omnect_validate_update_failed_file="${omnect_run_dir}/omnect_validate_update_failed"
     local omnect_validate_update_failed_flag="$(get_bootloader_env_var omnect_validate_update_failed)"
     local omnect_validate_update_file="${omnect_run_dir}/omnect_validate_update"
     local omnect_validate_update_flag="$(get_bootloader_env_var omnect_validate_update)"
 
+    [ -z "${omnect_device_service_user}" ] && msg "ERROR: user omnect_device_service not found in rootfs" && return 1
+    [ -z "${omnect_device_service_group}" ] && msg "ERROR: group omnect_validate_update not found in rootfs" && return 1
+
+    run_cmd mkdir -p ${omnect_run_dir} \
+    && run_cmd chown ${omnect_device_service_user}:${omnect_device_service_group} ${omnect_run_dir} \
+    && run_cmd chmod 775 ${omnect_run_dir}
+
     if [ -n "${omnect_validate_update_flag}" ] && [ -n "${omnect_validate_update_failed_flag}" ]; then
         msg_fatal "both flags are set \"omnect_validate_update\" and \"omnect_validate_update_failed\""
         return 1
-    elif [ -n "${omnect_validate_update_flag}" ] || [ -n "${omnect_validate_update_failed_flag}" ]; then
-        omnect_device_service_user="$(get_id omnect_device_service ${ROOTFS_DIR}/etc/passwd)"
-        omnect_validate_update_group="$(get_id omnect_validate_update ${ROOTFS_DIR}/etc/group)"
-        [ -z "${omnect_device_service_user}" ] && msg "ERROR: user omnect_device_service not found in rootfs" && return 1
-        [ -z "${omnect_validate_update_group}" ] && msg "ERROR: group omnect_validate_update not found in rootfs" && return 1
-        run_cmd mkdir -p ${omnect_run_dir} \
-        && run_cmd chown ${omnect_device_service_user}:${omnect_validate_update_group} ${omnect_run_dir} \
-        && run_cmd chmod 750 ${omnect_run_dir}
     fi
 
     # omnect update workflow handling
     if [ -n "${omnect_validate_update_flag}" ]; then
         run_cmd touch ${omnect_validate_update_file} \
-        && run_cmd chown ${omnect_device_service_user}:${omnect_validate_update_group} ${omnect_validate_update_file} \
-        && run_cmd chmod 600 ${omnect_validate_update_file}
+        && run_cmd chown ${omnect_device_service_user}:${omnect_device_service_group} ${omnect_validate_update_file} \
+        && run_cmd chmod 644 ${omnect_validate_update_file}
```

Comment:
what makes extension of access rights necessary?

Reply:
@JanZachmann patches in iot-hub-device-update